interfaces/builtin/autopilot.go

```diff
@@ -0,0 +1,58 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const autopilotIntrospectionPlugAppArmor = `
+# Description: Allows an application to be introspected and export its ui
+# status over dbus
+
+dbus (send, receive)
+    bus=session
+    path=/com/canonical/Autopilot/**
+    interface=org.freedesktop.DBus.Introspectable
+    member=Introspect
+    peer=(label=unconfined),
+dbus (send)
+    bus=session
+    path=/com/canonical/Autopilot/Introspection
+    interface=com.canonical.Autopilot.Introspection
+    member=GetVersion
+    peer=(label=unconfined),
+dbus (send)
+    bus=session
+    path=/com/canonical/Autopilot/Introspection
+    interface=com.canonical.Autopilot.Introspection
+    member=GetState
+    peer=(label=unconfined),
+`
+
+// NewAutopilotIntrospectionInterface returns a new "autopilot-introspection"
+// interface.
+func NewAutopilotIntrospectionInterface() interfaces.Interface {
+	return &commonInterface{
+		name: "autopilot-introspection",
```

Comment:
"autopilot-introspection" is a bit long. Don't change anything yet, but we want @niemeyer's input on the name. Gustavo, unlike other interfaces, this interface is about letting an unconfined process talk to the snap that plugs this interface so the unconfined process may gather the information necessary to drive the snap's UI. As such, our 'observe' and 'control' nomenclature doesn't really fit. I'm not opposed to "autopilot-intropection" per se (mostly because I don't have a better alternative), but at least wanted your feedback.